### ISSUE TITLE
pointsWithinPolygon excludes points that overlap with previous results

### ISSUE THREAD

**Issue by bahambalam** [Turfjs/turf#2150]
In my Leaflet app, the user can draw a rectangle/polygon to select points that fall within the drawn shape. They can draw a new shape, which will overwrite the old one and give them a new selection. I'm using pointsWithinPolygon for this, and it works great EXCEPT when the user selects specific points more than once. 

**Example:** 
Points I'm selecting from, shown as orange or yellow points 
![bug_pointsWithin1](https://user-images.githubusercontent.com/36213594/126558657-3dac9c22-d308-4af0-872b-5249cbdad7c5.png)

Drawing a selection box 
![bug_pointsWithin2](https://user-images.githubusercontent.com/36213594/126558718-ee35f346-0bd6-4c4b-aa08-8f8bb21bdf19.png)

The results of that selection, shown as black-colored points
![bug_pointsWithin3](https://user-images.githubusercontent.com/36213594/126558779-ed1b0360-1dfa-4f3c-b33d-26fde7e199af.png)

The results of a second selection. Points that were selected last time are excluded from the selection results this time, even though they should be included because they fall within the selection area.
![bug_pointsWithin4](https://user-images.githubusercontent.com/36213594/126558865-17fa862c-d36e-434c-a6a8-4823821c27f1.png)

If I draw the same exact shape twice, the first one returns normal results, and the second one returns zero results. Any idea how to fix this? I'm using Turf version 6.3.0. Thanks!

**Code:** 
selectedPts = turf.pointsWithinPolygon(turf.points(lyrPointsCoords), turf.polygon([drawnLyrCoords]));

**Comment by stebogit** [comment]
@bahambalam this really seems an issue related to the logic defining your `lyrPointsCoords` array.
Are you sure that array contains the entire set of points in you map, when is passed to the script calculating `selectedPts`, every time the selection polygon is drawn?

**Comment by bahambalam** [comment]
Thanks for responding. I was perplexed because I had made sure that the lyrPointsCoords array had the same number of points every time. However, when I investigated further, I realized another part of my code was swapping the positions of latitude and longitude for selected points (which was necessary so that a different Leaflet function would work properly). That's why the selected points weren't included in subsequent selections -- their lat/longs were reversed, and thus pointsWithinPolygon correctly identified them as not being within the drawn area. I have fixed it. I appreciate you helping me figure this out.
